iar: remove nxp duplicate definitions

0xc0170 opened the pull request:
### Description

<!-- 
    Required
    Add here detailed changes summary, testing results, dependencies 
    Good example: https://os.mbed.com/docs/latest/reference/workflow.html (Pull request template)
-->

Fixes #7634

Removing 2 duplicates (one was even using wrong device - instead of ....5 was .....12)

### Pull request type

<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [X] Fix
    [ ] Refactor
    [ ] New target
    [ ] Feature
    [ ] Breaking change

0xc0170 committed:
iar: remove nxp duplicate definitions

Fixes #7634

0xc0170 commented:
@theotherjimmy / @adbridge  can you verify these duplicates?

adbridge reviewed (commented):
LGTM I see the incorrect duplicates are 
"MK22DN512xxx5": {
"OGChipSelectEditMenu": "MK22FN512xxx12\tFreescale MK22FN512xxx12"
and
 "MK22DN512xxx5": {
"OGChipSelectEditMenu": "MK22DN512xxx5\tNXP MK22DN512xxx5"

One 12 and one 5 as you say. Just need to double check that we are removing the correct one.

0xc0170 committed:
frdm-k22f: fix IAR definition

Based on the https://www.keil.com/boards2/nxp/frdm_k22f/ - it contains xxx12 MCU.
However IAR 7.80 has only xxx5 version, thus leaving NXP added version as xxx5.

0xc0170 commented:
> One 12 and one 5 as you say. Just need to double check that we are removing the correct one.

Fixed. Based on the CMSIS pack info, K22F board should contain xxx12 MCU thus corrected and xxx5 removed as its not used anywhere.

adbridge reviewed (approved):
Looks more consistent now.

0xc0170 commented:
IAR 7.80 has only xxx5 device (DNxxx5) thus using that one in case of IAR. Tested locally

0xc0170 committed:
iar: kl43 - use nxp MCU definition

0xc0170 commented:
uvision and IAR works. However for uvision, device_name seems to be deprecated (shows me deprecated in the device selection in uvision).

mmahadevan108 reviewed (approved):
Thanks, looks good.

0xc0170 commented:
/morph build

cmonr commented:
@0xc0170 Could you locally confirm the error? It's possible this might be related to IAR issues we've been seeing, but it would be good to confirm first.

0xc0170 commented:
> @0xc0170 Could you locally confirm the error? It's possible this might be related to IAR issues we've been seeing, but it would be good to confirm first.

I did locally, but will do again the same as exporters do. It looks like this is error in this PR. I'll check

0xc0170 committed:
iar: MK22F use xx12 Freescale for 7.80.x

This fixes the failure for IAR regarding floating point.

0xc0170 commented:
I was able to reproduce the failure. It should be now fixed!

0xc0170 commented:
/morph build
